test/connect.js

```diff
@@ -418,39 +418,65 @@ QUnit.module('@connect with ObserveObject', () => {
 
 	});
 
-	QUnit.asyncTest('should only use splice on arrays passed into children not objects', (assert) => {
-		class FooList extends ObserveArray {
-			constructor(initialValue, bar) {
-				super(initialValue);
-				this.bar = bar;
+	QUnit.test('should preserve non-numeric key properties on observable arrays passed as props', (assert) => {
+		assert.expect(2);
+		const done = assert.async();
+
+		class ParentViewModel extends ObserveObject {
+			constructor(props) {
+				super(props);
+				this.observableArray = new ObserveArray([ 1 ]);
+				this.observableArray.foo = 'foo';
+				setTimeout((value) => {
+					const newObservableArray = new ObserveArray([ 2 ]);
+					newObservableArray.foo = 'bar';
+					this.observableArray = newObservableArray;
+					assert.equal(getTextFromElement(parentDiv), 'bar', 'bar');
+					done();
+				}, 10);
 			}
 		}
-		
-		function makeFooArray(bar) {
-			const fooArray = [ 1, 2, 3 ];
-			fooArray.bar = bar;
-			return fooArray;
-		}
 
-		class ParentStore extends ObserveObject {
+		const Parent = connect(ParentViewModel)(({ observableArray }) => (
+			<Child observableArray={observableArray} />
+		));
+		const Child = connect(EmptyViewModel)(({ observableArray }) => (
+			<div>{observableArray.foo}</div>
+		));
+
+		const parentInstance = ReactTestUtils.renderIntoDocument(<Parent />);
+		const parentDiv = ReactTestUtils.findRenderedDOMComponentWithTag(parentInstance, 'div');
+		assert.equal(getTextFromElement(parentDiv), 'foo', 'foo');
+	});
+
+	QUnit.test('should not preserve non-numeric key properties on non-observable arrays passed as props', (assert) => {
```

Comment:
Yeah makes sense, no table flipping necessary lol!

Personally I think we should preserve non-numeric key properties on non-observable arrays, it's perfectly reasonable for someone to put some extra info on an array, and who are we to mess with that. After looking into it, it seems to be a failing on `can-diff`'s part. `can-diff.deep()` isn't telling us that the non-numeric property has changed. 

Why exactly are we diff-ing the props anyway, is there some reason we can't just pass the new prop through?

Reply:
After talking with christopher on Friday, he suggested that we should check to see if something is observable and just copy it over if its not the same, rather than diffing inside of it.
So if you just pass a regular object or array, it will get diffed/merged. if you pass an observable, it will jut be replaced (assuming it changes).

This would fix the current problem that we're having with the demos, but still doesn't address the problem with normal arrays. 

Would using `canDiff.map` in this way work in the long run since it's not checking nested lists or maps?